app/controllers/devise_token_auth/sessions_controller.rb

```diff
@@ -12,25 +12,19 @@ def create
       # Check
       field = (resource_params.keys.map(&:to_sym) & resource_class.authentication_keys).first
 
-      @resource = nil
-      if field
-        q_value = resource_params[field]
-
-        if resource_class.case_insensitive_keys.include?(field)
-          q_value.downcase!
-        end
-
-        q = "#{field.to_s} = ? AND provider='email'"
-
-        if ActiveRecord::Base.connection.adapter_name.downcase.starts_with? 'mysql'
-          q = "BINARY " + q
-        end
+      q_value = resource_params[field]
+      if resource_class.case_insensitive_keys.include?(field)
+        q_value.downcase!
+      end
 
-        @resource = resource_class.where(q, q_value).first
+      set_resource(field, q_value)
+      unless @resource.present?
```

Comment:
Anything below here probably isnt needed (besides the `def resource_params...end` block)